sherpa/astro/tests/test_astro.py

```diff
@@ -47,759 +53,766 @@
 # has_xspec = has_package_from_list("sherpa.astro.xspec")
 
 
+@pytest.fixture
+def fix_xspec(clean_astro_ui, hide_logging):
+    """As of XSPEC 12.11.1 it's useful to fix abundance/cross sections
+    rather than rely on reading from the user's directory
+
+    This requires XSPEC support.
+    """
+
+    # As of XSPEC 12.10.1, it is safest to explicitly set
+    # the state to a known value. For now just pick the
+    # "easy" settings.
+    #
+    abund = xspec.get_xsabund()
+    xsect = xspec.get_xsxsect()
+
+    xspec.set_xsabund('angr')
+    xspec.set_xsxsect('bcmc')
+
+    yield
+
+    xspec.set_xsabund(abund)
+    xspec.set_xsxsect(xsect)
+
+
+@requires_data
+@requires_fits
+@requires_xspec
+@pytest.mark.parametrize('parallel', [False, True])
+def test_pha_intro(parallel, run_thread, fix_xspec):
+
+    # Currently there's no test of covarerr
+    def cmp_pha_intro(fit_result, p1, covarerr):
+        assert fit_result.statval == approx(37.9079, rel=1e-4)
+        assert fit_result.rstat == approx(0.902569, rel=1e-4)
+        assert fit_result.qval == approx(0.651155, rel=1e-4)
+        assert fit_result.nfev == 22
+        assert fit_result.numpoints == 44
+        assert fit_result.dof == 42
+        p1.gamma.val == approx(2.15852, rel=1e-4)
+        p1.ampl.val == approx(0.00022484, rel=1e-4)
+
+        assert ui.calc_photon_flux() == approx(0.000469964, rel=1e-4)
+        assert ui.calc_energy_flux() == approx(9.614847e-13, rel=1e-4)
+        assert ui.calc_data_sum() == approx(706.85714092, rel=1e-4)
+        assert ui.calc_model_sum() == approx(638.45693377, rel=1e-4)
+        assert ui.calc_source_sum() == approx(0.046996409, rel=1e-4)
+
+        calc = ui.eqwidth(p1, ui.get_source())
+        assert calc == approx(-0.57731725, rel=1e-4)
+
+        calc = ui.calc_kcorr([1, 1.2, 1.4, 1.6, 1.8, 2], 0.5, 2)
+        # Prior to fixing #619 the expected values were
```

Comment:
I have a slightly different feeling to some of these comments (they can help in identifying how the test has evolved, since occasionally we can end up changing a test over time such that it no-longer actually tests what it was originally intended to test), but that's not true here...